sql/core/src/main/scala/org/apache/spark/sql/catalog/interface.scala

```diff
@@ -222,3 +222,42 @@ class Function(
   }
 
 }
+
+/**
+ * A collation in Spark, as returned by the `listCollations` method defined in [[Catalog]].
+ *
+ * @param catalog name of the catalog that the collation belongs to.
+ * @param schema the schema that the collation belongs to.
+ * @param collationName name of the collation
+ * @param language language of the collation
+ * @param country country of the collation
+ * @param icuVersion icu version of the collation
+ * @param padAttribute pad attribute of the collation
+ * @param accentSensitivity accent sensitivity of the collation
+ * @param caseSensitivity case sensitivity of the collation
+ * @since 4.0.0
+ */
+class CollationMetadata(
```

Comment:
If we decide to not add listCollations to `Catalog` for now, we do not need this as well.

Reply:
Removed.